src/libraries/System.Net.Security/src/System/Net/Security/SslStreamCertificateContext.Linux.cs

```diff
@@ -252,6 +266,8 @@ private SslStreamCertificateContext(X509Certificate2 target, ReadOnlyCollection<
                     }
                 }
 
+                issuerHandles.AsSpan().Clear();
+                ArrayPool<IntPtr>.Shared.Return(issuerHandles);
                 ArrayPool<byte>.Shared.Return(rentedBytes);
                 ArrayPool<char>.Shared.Return(rentedChars.Array!);
                 GC.KeepAlive(TargetCertificate);
```

Comment:
Because we're touching a lot of objects .Handle without an upref/downref, there should be a GC.KeepAlive on the whole IntermediateCertificates collection (to prevent the GC from deciding that whole swathe of certificates can be finalized in a degenerate case).

Reply:
added `GC.KeepAlive(_privateIntermediateCertificates);`